flytekit/core/promise.py

```diff
@@ -1114,7 +1117,9 @@ def create_and_link_node_from_remote(
         upstream_nodes=upstream_nodes,
         flyte_entity=entity,
     )
-    ctx.compilation_state.add_node(flytekit_node)
+
+    if link_node:
```

Comment:
flag is maybe more correctly named as add_node_to_compilation_state or something...